### PR TITLE
fix(central): schema names with - should be allowed

### PR DESCRIPTION
**Pull request by mswertz**
needed because of 'catalogue-demo' schema (cannot now be created via UI while we use it for tests)

closes https://github.com/molgenis/molgenis-emx2/issues/4400

### CODE REVIEW

**Review by chinook25 — CHANGES_REQUESTED** [review]
Doesnt work: `test-test` fails. Regex needs to be: `/^(?!.* _|.*_ )[-a-zA-Z][-a-zA-Z0-9 _]{0,30}$`
(another `-` in between the last [])

File: apps/central/src/components/SchemaCreateModal.vue [hunk 1]
```diff
@@ -168,7 +168,7 @@ export default {
   },
   methods: {
     validate(name) {
-      const simpleName = constants.TABLE_NAME_REGEX;
+      const simpleName = constants.SCHEMA_NAME_REGEX;
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/central/src/components/SchemaCreateModal.vue:171]: I will check, should be check but if missing will add

**Comment by mswertz** [comment]
> Doesnt work: `test-test` fails. Regex needs to be: `/^(?!.* _|.*_ )[-a-zA-Z][-a-zA-Z0-9 _]{0,30}$` (another `-` in between the last [])

indeed, thx

File: apps/central/src/components/SchemaCreateModal.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/central/src/components/SchemaCreateModal.vue:171]: added

**Comment by mswertz** [comment]
> Doesnt work: `test-test` fails. Regex needs to be: `/^(?!.* _|.*_ )[-a-zA-Z][-a-zA-Z0-9 _]{0,30}$` (another `-` in between the last [])

Shouldn't it be `/^(?!.* _|.*_ )[a-zA-Z][-a-zA-Z0-9 _]{0,30}$` to prevent a start with -?

**Comment by chinook25** [comment]
> > Doesnt work: `test-test` fails. Regex needs to be: `/^(?!.* _|.*_ )[-a-zA-Z][-a-zA-Z0-9 _]{0,30}$` (another `-` in between the last [])
> 
> Shouldn't it be `/^(?!.* _|.*_ )[a-zA-Z][-a-zA-Z0-9 _]{0,30}$` to prevent a start with -?

If you want to prevent that, yes ;)

**Pull request merged**